test/git_repository.go

```diff
@@ -407,6 +394,15 @@ func (repo *GitRepository) FilesInBranches() (result DataTable, err error) {
 	return result, err
 }
 
+// HasBranchesOutOfSync indicates whether one or more local branches are out of sync with their remote
+func (repo *GitRepository) HasBranchesOutOfSync() (bool, error) {
+	res, err := repo.Shell.Run("git", "for-each-ref", "--format=%(refname:short) %(upstream:track)", "refs/heads")
```

Comment:
Very nice.

Reply:
Yes, our previous solution didn't show branches that have commits that the remote doesn't have. Found via the unit tests. This one works in all cases. Microsoft is adding a ton of usability features like this one to Git since they started using it for Windows and Office development.